cloud/shared/bin/doctor

```diff
@@ -0,0 +1,15 @@
+#! /usr/bin/env bash
+
+set -e
+
+export PROJECT_BASE_DIR="$(realpath "$(dirname "${BASH_SOURCE}")/../../../")"
+"${PROJECT_BASE_DIR}/cloud/shared/bin/validate_cloud_provider"
```

Comment:
mmm for this why are we validating the cloud provider here still?

Reply:
It influences which doctor checks run. Trying to prevent the situation where someone runs doctor, sees that everything looks right, then sets their cloud provider, runs setup, gets errors for not having the right CLI tools installed or configured.